File: src/main/java/com/example/chatService/repository/PrivateMessageRepository.java
package com.example.chatService.repository;

import com.example.chatService.entity.PrivateMessage;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PrivateMessageRepository extends JpaRepository<PrivateMessage, Long> {
    List<PrivateMessage> findBySenderAndReceiverOrderBySentAtAsc(String sender, String receiver);
    List<PrivateMessage> findByReceiverAndSenderOrderBySentAtAsc(String receiver, String sender);

}

File: src/main/java/com/example/chatService/repository/PrivateMessageRepository.kt
package com.example.chatService.repository

import com.example.chatService.entity.PrivateMessage
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PrivateMessageRepository : JpaRepository<PrivateMessage, Long> {
    fun findBySenderAndReceiverOrderBySentAtAsc(sender: String, receiver: String): List<PrivateMessage>

    fun findByReceiverAndSenderOrderBySentAtAsc(receiver: String, sender: String): List<PrivateMessage>
}
